sentry/src/main/java/io/sentry/Sentry.java

```diff
@@ -33,7 +33,8 @@ private Sentry() {}
    *
    * @return the hub
    */
-  static @NotNull IHub getCurrentHub() {
+  @ApiStatus.Internal // exposed for the coroutines integration in SentryContext
+  public static @NotNull IHub getCurrentHub() {
```

Comment:
what's about exposing it in the `IHub` and using `HubAdapter` instead of `Sentry` static class? same for `setCurrentHub`

Reply:
Nvm, it should be under the sentry static class anyway

Reply:
It can be added to `HubAdapter` and call package-private methods on `Sentry` class, but it cannot be added to `IHub` as there is no concept of the "current hub" on the hub itself. 

If the goal would be to hide the "current hub" concept from the public `Sentry` class API, we can add it to `HubAdapter` - am I getting it right?

Reply:
yes, the goal would be to not expose as a public method, even with `@ApiStatus.Internal`, if it can be package-private it'd be better IMO.
wdyt @bruno-garcia ?

Reply:
`currentHub` lives in the Sentry static class, moving to `HubAdapter` would be strange though, maybe we just need to be sure that its well tested and resetting the static values correctly in tests

Reply:
`currentHub` can stay in Sentry class, we just make these methods package private in Sentry and expose them with `HubAdapter` - since users don't use HubAdapter public API won't be polluted

Reply:
Makes sense to have this here wit @Internal and to be honest it's probably not going to be internal forever since this use case is real and users might want to propagate the Sentry state by themselves. This is done in Python, Rust and other SDKs where the user is expected to set the hub at the right places where applicable.

Reply:
sounds good, thanks